### PR TITLE
UserDetails addition in awgment-commons

### PR DESCRIPTION
**Pull request by denduluriramasaimohanakhil**
dev: 
 lint:0
  coverage:85%
utilityAddition:
  lint:0
  coverage:85%

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -31,6 +31,8 @@ configurations {
 
 
 dependencies {
+	implementation 'org.springframework.boot:spring-boot-starter-web:2.6.6'
+	implementation 'org.springframework.boot:spring-boot-starter-data-mongodb:2.6.6'
```
Comment (VaibhavTechsophy) [inline build.gradle]: do we need mongodb dependency? why?

File: src/main/java/com/techsophy/tsf/commons/acl/ACLEvaluatorImpl.java [hunk 2]
```diff
@@ -59,7 +62,7 @@ public ACLDecision getUpdate(String aclId,String token,Map<String,?> context)
 
     public ACLDecision getDelete(String aclId,Map<String,?> context)
     {
-        return   evaluateACL(aclId,tokenUtils.getTokenFromContext(), context).getDelete();
+        return   evaluateACL(aclId,tokenUtils.getToken().orElseThrow(), context).getDelete();
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/commons/acl/ACLEvaluatorImpl.java:65]: throw what?

File: src/main/java/com/techsophy/tsf/commons/acl/CommonsApplication.java [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.techsophy.tsf.commons.user.UserDetails;
+
 public class CommonsApplication
 {
-	public static void main(String[] args)
-    {
+	public static void main(String[] args) throws JsonProcessingException {
         //empty main method to support jar execution
+        UserDetails userDetails =new UserDetails("https://api-gateway.techsophy.com/api");
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/commons/acl/CommonsApplication.java]: pls remove this code

File: src/main/java/com/techsophy/tsf/commons/user/UserDetails.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        }
+        return authentication.getPrincipal();
+    }
+
+    public Optional<String> getIssuer()
+    {
+        Object principal = getPrincipal();
+        if (principal instanceof OAuth2User)
+        {
+            return Optional.ofNullable(((OAuth2User) principal).getName());
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/commons/user/UserDetails.java]: why is name being returned as issuer?

File: src/main/java/com/techsophy/tsf/commons/user/UserFormDataDefinition.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.techsophy.tsf.commons.user;
+
+import lombok.Data;
+
+@Data
+public class UserFormDataDefinition
+{
+    private UserData userData;
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/commons/user/UserFormDataDefinition.java]: this does not look complete

File: src/test/java/com/techsophy/tsf/commons/utils/TokenUtilsTest.java [hunk 6]
```diff
@@ -1,65 +1,223 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({MockitoExtension.class})
 class TokenUtilsTest
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/commons/utils/TokenUtilsTest.java]: pls rename this file

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]
lets discuss

File: src/main/java/com/techsophy/tsf/commons/user/UserDetails.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+            return Optional.ofNullable(String.valueOf(jwt.getIssuer()));
+        }
+    }
+
+    public Optional<String> getToken()
+    {
+        Object principal =getPrincipal();
+        if (principal instanceof OAuth2User)
+        {
+            return Optional.ofNullable(((OAuth2User) principal).getName());
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/commons/user/UserDetails.java]: why are we returning name for token

File: build.gradle (hunk 1 above)
Comment (denduluriramasaimohanakhil) [reply build.gradle]: Yes sir removing dependencies

File: src/main/java/com/techsophy/tsf/commons/acl/CommonsApplication.java (hunk 3 above)
Comment (denduluriramasaimohanakhil) [reply src/main/java/com/techsophy/tsf/commons/acl/CommonsApplication.java]: yes removing unwanted code now

File: src/main/java/com/techsophy/tsf/commons/user/UserDetails.java (hunk 4 above)
Comment (denduluriramasaimohanakhil) [reply src/main/java/com/techsophy/tsf/commons/user/UserDetails.java]: correcting it

File: src/test/java/com/techsophy/tsf/commons/utils/TokenUtilsTest.java (hunk 6 above)
Comment (denduluriramasaimohanakhil) [reply src/test/java/com/techsophy/tsf/commons/utils/TokenUtilsTest.java]: yes renaming

File: src/main/java/com/techsophy/tsf/commons/user/UserDetails.java [hunk 8]
```diff
@@ -105,7 +91,7 @@ public UserFormDataDefinition getUserDetails() throws JsonProcessingException
...
...hbmFnZS11c2VycyIsImF3Z21lbnQtY2hlY2tsaXN0LWFsbCIsImF3Z21lbnQtbW9kZWxlci10ZXN0IiwiYXdnbWVudC1jaGVja2xpc3QtYWRtaW4iLCJhd2dtZW50LWFjY291bnQtYWxsIiwiYXdnbWVudC1ydWxlLWVuZ2luZS1hbGwiLCJhd2dtZW50LXN1cGVyc2V0LWFscGhhIiwiYXdnbWVudC13b3JrZmxvdy1lbmdpbmUtY3JlYXRlLW9yLXVwZGF0ZSIsImF3Z21lbnQtbW9kZWxlci1ydWxlIiwiYXdnbWVudC1jYXNlLWFsbCIsImF3Z21lbnQtbW9kZWxlci13b3JrZmxvdyIsImF3Z21lbnQtbWFuYWdlLWNvbXBvbmVudHMiLCJhd2dtZW50LWZvcm0tYWNsLWFsbCIsImF3Z21lbnQtbm90aWZpY2F0aW9uLWFsbCIsIkFkbWluIiwiYXdnbWVudC1ydWxlLWFsbCIsImF3Z21lbnQtbWFuYWdlLXRoZW1lcyIsImF3Z21lbnQtd29ya2Zsb3ctZW5naW5lLWRlbGV0ZSIsImF3Z21lbnQtc2NoZWR1bGVyLWFsbCIsImF3Z21lbnQtdXRpbC1hbGwiLCJhd2dtZW50LWNoZWNrbGlzdC1lbmdpbmUtYWxsIiwiYXdnbWVudC1kb2N1bWVudC1jb252ZXJ0ZXItYWxsIiwiYXdnbWVudC1zdXBlcnNldC1tYW5hZ2VyIiwiYXdnbWVudC1zdXBlcnNldC1zcWxfbGFiIiwiYXdnbWVudC1zdXBlcnNldC1ncmFudGVyIiwiYXdnbWVudC1hY2wtcmVhZCJdfQ.K4Fi7tun3yAEGiSovtX2zMdcYXwr-qsZ7mrRW5LIBNYagaf2aoE4WYEsX1GCV7Yn7Wj65MFLB7Q67dQ_eadBPX9Sw_fTORbCrYCj_NLcKY8l8PsSlWcptfbqpXdi0muRvU-PwYysSwBHzl0a27cPYvbMhDG4bWm8Lkkh2prMnLvA6YzuKcQb0LpCEgoRCfcLWgcCNbWVZzWnjCGuzOHbBqnFEgbOqiCSjCWTLA0JTFwjpq-NPh2ehsPr7ikSVfo5-FE-GwNUwyAybn_pRrMpEUXXeiJNclk1x4Y6LgXS7KvIscng2sHJAUtOxVK7nbwN-rCXnjn3xsp_vARcTr9QlA")
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/commons/user/UserDetails.java]: what is this?

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]
Please review yourself first before raising merge request

**Review by VaibhavTechsophy — APPROVED** [review]

**Pull request merged**